src/components/Admin/ManageSensors/SensorTable/SensorTable.tsx

```diff
@@ -25,6 +37,113 @@ const SensorTable: ({title, sensors}: SensorTableProps) => JSX.Element = ({
 }: SensorTableProps) => {
   const {t} = useTranslation('sensors');
 
+  /**
+   * Get the human readable error name and explanation
+   * @param error - the error that occurred
+   * @returns a tuple of the human readable error name and explanation
+   */
+  function getSensorReadingErrorInfo(
+    error: SensorReadingErrors
+  ): [name: string, explanation: string] {
+    switch (error) {
+      case SensorReadingErrors.ReadingNotReceived:
+        return [
+          t('sensorErrors.readingNotReceived.heading'),
+          t('sensorErrors.readingNotReceived.explanation'),
+        ];
+      case SensorReadingErrors.NoHumidityReading:
+        return [
+          t('sensorErrors.noHumidity.heading'),
+          t('sensorErrors.noHumidity.explanation'),
+        ];
+      case SensorReadingErrors.IncompleteSensorReading:
+        return [
+          t('sensorErrors.incompleteReading.heading'),
+          t('sensorErrors.incompleteReading.explanation'),
+        ];
+      case SensorReadingErrors.ChannelsDiverged:
+        return [
+          t('sensorErrors.channelsDiverged.heading'),
+          t('sensorErrors.channelsDiverged.explanation'),
+        ];
+      case SensorReadingErrors.ChannelADowngraded:
+        return [
+          t('sensorErrors.channelADowngraded.heading'),
+          t('sensorErrors.channelADowngraded.explanation'),
+        ];
+      case SensorReadingErrors.ChannelBDowngraded:
+        return [
+          t('sensorErrors.channelBDowngraded.heading'),
+          t('sensorErrors.channelBDowngraded.explanation'),
+        ];
+      default:
+        // Unknown error
+        return ['', ''];
+    }
+  }
+
+  /**
+   * Get the error tags for all sensor errors of a sensor
+   * @param sensor - the sensor to check the sensor errors of
+   * @returns an array of the error tags that are true for that sensor
+   */
+  function getSensorErrorTags(sensor: Sensor): JSX.Element[] {
+    const errorTags: JSX.Element[] = [];
+    sensor.sensorReadingErrors.forEach((value, index) => {
+      if (value) {
+        const [name, explanation] = getSensorReadingErrorInfo(index);
```

Comment:
This seems to be passing a number `index` into `getSensorReadingErrorInfo`, but `getSensorReadingErrorInfo` expects an input parameter `error`. I see how this works, but it's not clear upon a first read. Consider changing this to `error_index` or `error_value` and then edit the documentation for `getSensorReadingErrorInfo` to mention that the input error is a number.

Reply:
I think updating the error array implementation to a list of integers that correspond to errors (instead of an array of booleans) will make this more clear